com.unity.multiplayer.mlapi/Runtime/Core/NetworkedBehaviour.cs

```diff
@@ -14,19 +14,105 @@
 using MLAPI.Profiling;
 using MLAPI.Reflection;
 using MLAPI.Security;
+using MLAPI.Serialization;
 using MLAPI.Serialization.Pooled;
 using MLAPI.Spawning;
 using BitStream = MLAPI.Serialization.BitStream;
-using MLAPI.Serialization;
 using Unity.Profiling;
 
+#if UNITY_EDITOR
+using System.Runtime.CompilerServices;
+[assembly: InternalsVisibleTo("Unity.Multiplayer.MLAPI.Editor.CodeGen")]
+#endif // UNITY_EDITOR
+
 namespace MLAPI
 {
     /// <summary>
     /// The base class to override to write networked code. Inherits MonoBehaviour
     /// </summary>
     public abstract partial class NetworkedBehaviour : MonoBehaviour
     {
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal enum NExec
+        {
+            None = 0,
+            Server = 1,
+            Client = 2
+        }
+
+        const String StandardRPC_Channel = "STDRPC";
+
+#pragma warning disable 414
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal NExec __nexec = NExec.None;
+#pragma warning restore 414
+
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal BitWriter BeginSendServerRpc(ServerRpcSendParams sendParams, bool isReliable)
+        {
+            // @mfatihmar (Unity) Begin: Temporary, placeholder implementation
+            RPCQueueManager rpcQueueMananger = NetworkingManager.Singleton.GetRPCQueueManager();
+            if(rpcQueueMananger != null)
+            {
+                var writer = rpcQueueMananger.BeginAddQueueItemToOutboundFrame(RPCQueueManager.QueueItemType.ServerRPC, Time.realtimeSinceStartup, StandardRPC_Channel,0, NetworkingManager.Singleton.ServerClientId,null);
+
+                writer.WriteBit(false); // Encrypted
+                writer.WriteBit(false); // Authenticated
+                writer.WriteBits(MLAPIConstants.MLAPI_STD_SERVER_RPC, 6); // MessageType
```

Comment:
Magical '6'

Reply:
Yes, this actually was pretty much the way it was being done originally...
Two bits for the encrypted and authenticated flags and the remaining 6 bits for the message type yielding 8 bits for the first byte of the MLAPI header.